### PR TITLE
Added InstantMapper and InstantArgument with Factory

### PR DESCRIPTION
**Pull request by softarn**
Added InstantArgumentFactory. A bit confused why no one have added this yet. So please tell me if there is a good reason to not do this since I'm currently using it in my own projects.

### CODE REVIEW

File: dropwizard-java8-jdbi/src/main/java/io/dropwizard/java8/jdbi/args/InstantArgument.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private Instant instant;
+
+    protected InstantArgument(Instant instant) {
+        this.instant = instant;
+    }
+
+    @Override
+    public void apply(int position, PreparedStatement statement, StatementContext ctx) throws SQLException {
+        if (instant != null) {
+            statement.setTimestamp(position, new Timestamp(instant.toEpochMilli()));
```
Comment (nickbabcock) [inline dropwizard-java8-jdbi/src/main/java/io/dropwizard/java8/jdbi/args/InstantArgument.java]: Should it be [`Timestamp.from(instant)`](https://docs.oracle.com/javase/8/docs/api/java/sql/Timestamp.html#from-java.time.Instant-)?

Comment (softarn) [inline dropwizard-java8-jdbi/src/main/java/io/dropwizard/java8/jdbi/args/InstantArgument.java]: Yes, I saw this way and just wanted to try it. Forgot to switch back.
Thanks

**Comment by nickbabcock** [comment]
Does it make sense to define `InstantArgumentMapper`? `LocalDate` and `LocalDateTime` both have an `Argument`, `ArgumentFactory`, and `Mapper` defined for them. 

Other than that, it looks good to me! :+1:

**Comment by softarn** [comment]
Updated

**Comment by joschi** [comment]
@softarn I think you forgot to register `InstantMapper` in `DBIFactory`.

**Comment by softarn** [comment]
Updated

**Comment by joschi** [comment]
LGTM. :+1:

Thanks for your contribution!

**Pull request merged**